layers/+tools/eaf/README.org

```diff
@@ -0,0 +1,89 @@
+#+TITLE: eaf layer
+# Document tags are separated with "|" char
+# The example below contains 2 tags: "layer" and "web service"
+# Avaliable tags are listed in <spacemacs_root>/.ci/spacedoc-cfg.edn
+# under ":spacetools.spacedoc.config/valid-tags" section.
+#+TAGS: layer|web service
+
+# The maximum height of the logo should be 200 pixels.
+# [[img/eaf.png]]
+
+# TOC links should be GitHub style anchors.
+* Table of Contents                                        :TOC_4_gh:noexport:
+- [[#description][Description]]
+  - [[#features][Features:]]
+- [[#install][Install]]
+- [[#usage][Usage]]
+    - [[#configuring-keybindings][Configuring keybindings]]
+- [[#key-bindings][Key bindings]]
+  - [[#all-eaf-applications][All EAF-applications]]
+  - [[#browser][Browser]]
+      - [[#global][Global]]
+      - [[#local][Local]]
+
+* Description
+  This layer adds support for the [[https://github.com/manateelazycat/emacs-application-framework][Emacs Application Framework (EAF)]].
+
+** Features:
+  - Browse using a full-fledged browser within Emacs
+  - PDF viewer (with continuous scroll)
+  - Video player
+  - Image viewer
+  - See [[https://github.com/manateelazycat/emacs-application-framework#launch-eaf-applications][EAF documentation]] for many more features
+
+* Install
+  To use this configuration layer, add it to your =~/.spacemacs=. You will need to
+  add =eaf= to the existing =dotspacemacs-configuration-layers= list in this
+  file.
+
+  Subsequently install the required dependencies with =SPC SPC
+  install-eaf-dependencies=. [fm:: This command only works for Ubuntu and Fedora
+  based distributions. Users of Arch (based) distributions can run the script
+  manually in their terminal. If the script does not work for you then [[https://github.com/manateelazycat/emacs-application-framework#dependency-list][install
+  the required dependencies manually]].
+
+* Usage
+
+*** Configuring keybindings
+  This layer implements some tricks that are required to make EAF behave as
+  desired for Spacemacs. Therefore it is recommended to keep the default
+  (major-mode-)leader keybindings. Also due to these tricks, ~SPC SPC~ does not
```

Comment:
Can't we get this into the layer already? I have to say adding a layer which does not support leader key mechanisms would behave unexpectedly for most users I assume.

Reply:
So I think my text in the documentation was not clear enough. I hope my change improved its meaning.
The problem is that EAF keybindings interfere with evil keybindings. Therefore the EAF developers created an eaf-evil.el package, but I found it not compatible enough with Spacemacs. So still using their eaf-evil.el package, I added some modifications to "emulate" Spacemacs behavior. So for the user it works exactly how they expect from Spacemacs (except for the `SPC SPC` but for that they can just use `M-x`. EAF comes with custom eaf keybinding functions that somehow commincate the keybindings to the python part of EAF. You can read more about my modifications [here](https://github.com/manateelazycat/emacs-application-framework/pull/500) and [here](https://github.com/manateelazycat/emacs-application-framework/issues/498). So making EAF work the Spacemacs way is the strong point of this layer. Getting the keybindings right is not trivial.